indexdata.go

```diff
@@ -45,6 +45,17 @@ type indexData struct {
 	boundariesStart uint32
 	boundaries      []uint32
 
+	symbolNames []byte
+	symbolIndex []uint32
+
+	symbolMetaData      []byte
+	symbolMetaDataIndex []uint32
+
+	symbolNgrams      map[ngram][]uint32
+	symbolRuneOffsets []uint32
+	symbolEndRunes    []uint32
+	symbolFileOffsets []uint32
```

Comment:
Random idea, could we not just rely on what zoekt was storing before (symbol offsets + lengths), but in the matching process remove candidates which do not fit into a symbol offset and length? We then store offset -> kind and hydrate that in? This may be much less code, and wouldn't inflate our index as much.

It is also concerning to me how much extra data it looks like we are storing in memory here. These slices seem like they would be very large, much larger than any other slice here (eg everything else is bounded by document size / samples offsets (for utf8). This is bounded by number of symbols which can be many multiples higher than document count).